tools/decode.py

```diff
@@ -45,92 +53,51 @@ def run(file, **kwargs):
         # Work out our original protobuf type
         pb_type = message.__module__.split('.')[:-1]
         pb_type.append(message.__name__)
-        pb_type = '.'.join(pb_type).encode('utf-8')
-        pb_hash = '' # TODO swap to using XXHASH
-        # pb_hash = mmh3.hash_bytes(pb_type, 0x4e55436c, True)
-
+        pb_type = '.'.join(pb_type)
+        # Reverse to little endian
+        pb_hash = bytes(reversed(xxhash.xxh64(pb_type, seed=0x4e55436c).digest()))
         parsers[pb_hash] = (pb_type, message)
 
-    servo_health = {}
+        # We can also decode NUsight<> wrapped packets
+        nusight_type = 'NUsight<{}>'.format(pb_type)
+        # Reverse to little endian
+        nusight_hash = bytes(reversed(xxhash.xxh64(nusight_type, seed=0x4e55436c).digest()))
+        parsers[nusight_hash] = (nusight_type, NUsightDecoder(message))
+
 
     # Now open the passed file
     with gzip.open(file, 'rb') if file.endswith('nbz') or file.endswith('.gz') else open(file, 'rb') as f:
```

Comment:
It would be nice if there was either:

- a link to the definition of the file format
- or paste the definition I have in NUsight2 if one does not exist 😛